clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -913,7 +913,8 @@ private boolean isProtocolTypeInconsistent(String protocolType) {
     }
 
     private boolean isProtocolNameInconsistent(String protocolName) {
-        return protocolName != null && !protocolName.equals(generation().protocolName);
+        return protocolName != null && generation() != Generation.NO_GENERATION
+                   && !protocolName.equals(generation().protocolName);
```

Comment:
@guozhangwang @abbccdda Shouldn't we synchronise this or use a local reference of the generation to be 100% safe?